types/types.go

```diff
@@ -47,9 +47,12 @@ type ImageDestination interface {
 // Image is a Docker image in a repository.
 type Image interface {
 	// ref to repository?
+	// GetManifest is like ImageSource.GetManifest, but the result is cached; it is OK to call this however often you need.
+	GetManifest() ([]byte, error)
+	// GetSignatures is like ImageSource.GetSignatures, but the result is cached; it is OK to call this however often you need.
+	GetSignatures() ([][]byte, error)
```

Comment:
> but honestly the whole ImageSource/ImageDestination will need to be rethought if we ever support non-seekable tarball streams.

agree, maybe they could be embedded directly in images and are available only as a mean of developing new internal source|destination exposed by types.Image directy